### PR TITLE
fix(sign): fix alibabacloud signature error & add refine unit test

### PR DESCRIPTION
**Pull request by kaaanata**

### CODE REVIEW

File: aliSimple.go [hunk 1]
```diff
@@ -96,7 +96,7 @@ func (c *aliClient) getAliClientSche(cred *CredInfo, scheme string) (*aliClient,
 		APIHost: fmt.Sprintf(addrOfAPI, scheme),
 		reqMap: []vKey{
 			{key: "AccessKeyId", val: cred.AccKeyID},
-			{key: "Format", val: "json"},
```
Comment (kaaanata) [inline aliSimple.go:99]: https://help.aliyun.com/document_detail/29745.html#:~:text=%E7%9A%84%E7%B1%BB%E5%9E%8B%EF%BC%8C%E6%94%AF%E6%8C%81-,json%E4%B8%8Exml,-%E3%80%82%E9%BB%98%E8%AE%A4%E4%B8%BAXML

File: aliSimple.go [hunk 2]
```diff
@@ -189,7 +189,9 @@ func urlEncode(ins string) string {
 	str0 = strings.Replace(str0, "+", "%20", -1)
 	str0 = strings.Replace(str0, "*", "%2A", -1)
 	str0 = strings.Replace(str0, "%7E", "~", -1)
+
 	str0 = url.QueryEscape(str0)
+	str0 = strings.Replace(str0, "%26", "&", -1)
```
Comment (kaaanata) [inline aliSimple.go:194]: via  documentation sample, it should not escape "&"

https://help.aliyun.com/document_detail/29747.html#:~:text=3Dtestid%26Action%253DDescribeDomainRecords-,%26,-DomainName%253Dexample.com

File: aliSimple_test.go [hunk 3]
```diff
@@ -18,7 +18,8 @@ var cl0 = &aliClient{
 	APIHost: fmt.Sprintf(addrOfAPI, "https"),
 	reqMap: []vKey{
 		{key: "AccessKeyId", val: "testid"},
-		{key: "Acction", val: "DescribeDomainRecords"},
```
Comment (kaaanata) [inline aliSimple_test.go:21]: typo

File: aliSimple_test.go [hunk 4]
```diff
@@ -34,8 +35,14 @@ func Test_AliClintReq(t *testing.T) {
 	str := cl0.reqMapToStr()
 	t.Log("map to str:" + str + "\n")
 	str = cl0.reqStrToSign(str, "GET")
+
+	// validate sign string from doc: https://help.aliyun.com/document_detail/29747.html#:~:text=%E9%82%A3%E4%B9%88-,stringtosign
+	if str != "GET&%2F&AccessKeyId%3Dtestid&Action%3DDescribeDomainRecords&DomainName%3Dexample.com&Format%3DXML&SignatureMethod%3DHMAC-SHA1&SignatureNonce%3Df59ed6a9-83fc-473b-9cc6-99c95df3856e&SignatureVersion%3D1.0&Timestamp%3D2016-03-24T16%253A41%253A54Z&Version%3D2015-01-09" {
```
Comment (kaaanata) [inline aliSimple_test.go:40]: matching the documentation https://help.aliyun.com/document_detail/29747.html#:~:text=%E9%82%A3%E4%B9%88-,stringtosign

File: aliSimple.go (hunk 1 above)
Comment (viscropst) [reply aliSimple.go:99]: Could you tell me it is well tested with upper case or lower case? The example of your link was lower case. If yes, that was no problem.

Comment (kaaanata) [reply aliSimple.go:99]: image

According to the documentation, its upper case. And I've tested it in uppercase and works fine. Seems alibabacloud will automate resolve what ever upper case  or lower case

**Comment by viscropst** [comment]
Okay,It's LGTM.Thanks!

**Pull request merged**